src/sixLesson/DataCompression.java

```diff
@@ -0,0 +1,44 @@
+package sixLesson;
+
+public class DataCompression {
+
+
+    public static void main(String[] args) {
+
+        String str1 = "AAAAaaaBBBBBB\\12";
+        String value1 = encode(str1);
+        System.out.println("Output string1: "+value1);
+
+        String str2 = "AaaaBB2";
+        String value2 = encode(str2);
+        System.out.println("Output string2: "+value2);
+
+        String str3 = "AaaaBB\\2";
+        String value3 = encode(str3);
+        System.out.println("Output string3: "+value3);
+    }
+    public static String encode(String string) {
```

Comment:
Cannot process '/' character and digit characters. This implementation can only work with letters.